From 5b74db2eec3611fe9f1a75089797c5a99cd51fff Mon Sep 17 00:00:00 2001
From: Imran Zaman <imran.zaman@intel.com>
Date: Fri, 4 Sep 2015 14:32:34 +0300
Subject: [PATCH] Fixes possible overflow when copying the string

Bigger source buffer than dest buffer could
overflow when copying strings.
Source and destination buffer sizes are same now

Upstream-status: Submitted (http://lists.denx.de/pipermail/u-boot/2015-September/227067.html)

Signed-off-by: Imran Zaman <imran.zaman@intel.com>
---
 common/cli_simple.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/common/cli_simple.c b/common/cli_simple.c
index 00a8d2f..d8b40c9 100644
--- a/common/cli_simple.c
+++ b/common/cli_simple.c
@@ -258,7 +258,7 @@ int cli_simple_run_command(const char *cmd, int flag)
 
 void cli_simple_loop(void)
 {
-	static char lastcommand[CONFIG_SYS_CBSIZE] = { 0, };
+	static char lastcommand[CONFIG_SYS_CBSIZE + 1] = { 0, };
 
 	int len;
 	int flag;
-- 
1.9.1

